### DISCUSSION TITLE
How to run the server?

### DISCUSSION THREAD

**Discussion by neznayer** [node-saml/node-saml#376] [Q&A]
In the README, the only Node.js code is:
```js
const { SAML } = require("@node-saml/node-saml");

const options = {};
const saml = new SAML(options);
```

However, this code does effectively nothing, just creates some class instance. 
How to run the actual IdP server? 
`saml.listen(3000)` or something like that?

Best regards, and sorry for dumb question :)

**Comment by srd90** [comment]
If you are looking for usage examples then this question is duplicate of these:

* https://github.com/node-saml/node-saml/discussions/359
  * see answer https://github.com/node-saml/node-saml/discussions/359#discussioncomment-8765123
* https://github.com/node-saml/node-saml/issues/258
  * see answer https://github.com/node-saml/node-saml/issues/258#issuecomment-1460808497
* https://github.com/node-saml/node-saml/discussions/42

---

>How to run the actual IdP server?

If you are looking solution for IdP server (instead of library to implement e.g. SP side of web sso profile) node-saml is not for you.

If you do not know difference between IdP and SP then I suggests to study SAML 2.0 prior to implementing anything SAML related at all. Read through SAML 2.0 specs. List of specs is provided e.g. at the end of this comment: https://github.com/node-saml/node-saml/issues/326#issuecomment-1717221811

**Reply by neznayer** [reply]
Thank you! 
I'm really just starting with SAML. 
It's just from the Readme it is not clear, whether this is an IdP or SP side (for beginner like me:)
